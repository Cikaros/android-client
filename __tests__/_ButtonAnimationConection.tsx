import React from 'react';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import ButtonAnimationConnection from '../src/components/ButtonAnimationConnection';

const mockStore = configureStore();
const initialState = {}; // Define the initial state for the store
const store = mockStore(initialState);

jest.mock('../src/components/ButtonAnimationConnection', () => {
    return jest.fn().mockImplementation(({ toggleConnection }) => (
      <button onClick={toggleConnection} />
    ));
  });

describe('ButtonAnimationConnection', () => {

  it('should call toggle connection when button is pressed', () => {
    const toggleConnectionMock = jest.fn(); // Create a mock function for toggleConnection
  
    const wrapper = shallow(
      <Provider store={store}>
        <ButtonAnimationConnection ref={toggleConnectionMock} />
      </Provider>
    );
    const buttonAnimationConnection = wrapper.find(ButtonAnimationConnection).dive();
    
    // Simulate button press event
    buttonAnimationConnection.simulate('press');

    // Assert that toggleConnectionMock was called
    expect(toggleConnectionMock).toMatchSnapshot();
  });

  it('should call toggle connection when TouchableWithoutFeedback is pressed', () => {
    const toggleConnectionMock = jest.fn(); // Create a mock function for toggleConnection

    const wrapper = shallow(
      <Provider store={store}>
        <ButtonAnimationConnection ref={toggleConnectionMock} />
      </Provider>
    );

    const buttonAnimationConnection = wrapper.find(ButtonAnimationConnection).dive();
    const touchableWithoutFeedback = buttonAnimationConnection.find('TouchableWithoutFeedback');

    expect(wrapper).toMatchSnapshot();
    expect(touchableWithoutFeedback).toMatchSnapshot();
  });

});
