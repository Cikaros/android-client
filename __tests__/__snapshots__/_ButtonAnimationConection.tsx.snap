// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonAnimationConnection should call toggle connection when TouchableWithoutFeedback is pressed 1`] = `
<ContextProvider
  value={
    {
      "getServerState": undefined,
      "noopCheck": "once",
      "stabilityCheck": "once",
      "store": {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <mockConstructor />
</ContextProvider>
`;

exports[`ButtonAnimationConnection should call toggle connection when TouchableWithoutFeedback is pressed 2`] = `null`;

exports[`ButtonAnimationConnection should call toggle connection when button is pressed 1`] = `[MockFunction]`;
